packages/perseus-editor/src/widgets/plotter-editor.tsx

```diff
@@ -38,39 +37,57 @@ const editorDefaults = {
     snapsPerLine: 2,
 } as const;
 
-const widgetPropTypes = {
-    type: PropTypes.oneOf([BAR, LINE, PIC, HISTOGRAM, DOTPLOT]),
-    labels: PropTypes.arrayOf(PropTypes.string),
-    categories: PropTypes.arrayOf(
-        PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
-    ),
-
-    scaleY: PropTypes.number,
-    maxY: PropTypes.number,
-    snapsPerLine: PropTypes.number,
-
-    picSize: PropTypes.number,
-    pixBoxHeight: PropTypes.number,
-    picUrl: PropTypes.string,
-
-    plotDimensions: PropTypes.arrayOf(PropTypes.number),
-    labelInterval: PropTypes.number,
-    starting: PropTypes.arrayOf(PropTypes.number),
-    correct: PropTypes.arrayOf(PropTypes.number),
-    static: PropTypes.bool,
-    onChange: PropTypes.func,
-} as const;
+type EditingStates = "starting" | "correct";
+
+type Props = {
+    type: "bar" | "line" | "pic" | "histogram" | "dotplot";
+    labels: Array<string>;
```

Comment:
```suggestion
    labels: ReadonlyArray<string>;
```

Reply:
I defaulted to using `ReadonlyArray` when possible, but had to change some of these because they're getting directly mutated. I'll try again though, just in case.

Reply:
Ah, I wondered about that. That's unfortunate but I understand not wanting to go down a rabbit hole when just adding better types.